### PR TITLE
rename csv to lightning conventions & fix short value in CF_WRITE not encoded in csv

### PR DESCRIPTION
**Pull request by BugenZhao**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Please add a release note.
If you don't think this PR needs a release note then fill it with None.
```

### CODE REVIEW

File: components/backup_text/src/bin/rewrite/br_models.rs [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
+    pub original: String,
+    #[serde(rename = "L")]
+    pub lower: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+struct BrDbInfo {
+    pub id: i64,
+    #[serde(rename = "db_name")]
+    pub name: BrCiString,
```
Comment (NingLin-P) [inline components/backup_text/src/bin/rewrite/br_models.rs:16]: I think the original name is enough.

File: components/backup_text/src/bin/rewrite/main.rs [hunk 2]
```diff
@@ -151,22 +162,38 @@ async fn worker(opt: Opt) -> Result<()> {
...
-    };
-
-    write_message(&new_storage, META_FILE, new_meta)?;
-    info!("update meta file done");
+    // Post work
+    match mode {
+        RewriteMode::ToText | RewriteMode::ToSst => {
+            // update meta file for correct restoration
+            let new_meta = {
+                let mut meta = meta;
```
Comment (NingLin-P) [inline components/backup_text/src/bin/rewrite/main.rs]:
```suggestion
```

File: components/backup_text/src/bin/rewrite/main.rs [hunk 3]
```diff
@@ -151,22 +162,38 @@ async fn worker(opt: Opt) -> Result<()> {
...
+        RewriteMode::ToCsv => {
+            // cleanup
+            for file in mutated_file_map.into_values() {
+                let mut path = new_path.clone();
+                path.push(file.name);
+                if fs::metadata(&path)?.len() == 0 {
+                    fs::remove_file(&path)?;
+                }
+            }
+            info!("cleanup done");
```
Comment (NingLin-P) [inline components/backup_text/src/bin/rewrite/main.rs]: We can clean up empty files after `TextWriter::finish` through `TextWriter::get_size` and `TextWriter::cleanup`

File: components/backup_text/src/bin/rewrite/main.rs [hunk 4]
```diff
@@ -115,21 +115,32 @@ async fn worker(opt: Opt) -> Result<()> {
...
-        for file in files {
-            let table_info = table_info.clone();
+        let info = info_map.get(&table_id).expect("no table info found");
+        let total_files = files.len();
+        for (i, file) in files.into_iter().enumerate() {
+            let info = info.clone();
             let (dir, new_dir) = (path.clone(), new_path.clone());
             let name = file.get_name().to_owned();
+            let rename_to = (mode == RewriteMode::ToCsv).then(|| {
+                let width = total_files.to_string().len();
```
Comment (NingLin-P) [inline components/backup_text/src/bin/rewrite/main.rs]: How about move this line to L128

File: components/backup_text/src/rwer.rs [hunk 5]
```diff
@@ -141,37 +141,56 @@ impl TextWriter {
...
+                        .unwrap()
+                        .into_bytes()
+                }
+                (CF_WRITE, DataType::Record) => {
+                    kv_to_write(&mut self.ctx, key, val, &self.schema.columns).into_bytes()
+                }
+                (CF_WRITE, DataType::Index) => {
+                    index_kv_to_write(&mut self.ctx, key, val, &self.schema.columns).into_bytes()
+                }
+                (..) => unreachable!(),
```
Comment (NingLin-P) [inline components/backup_text/src/rwer.rs]:
```suggestion
                _ => unreachable!(),
```

File: components/backup_text/src/rwer.rs [hunk 6]
```diff
@@ -141,37 +141,56 @@ impl TextWriter {
...
+                        result.unwrap()
+                    } else {
+                        return Ok(());
+                    }
+                }
+                (CF_DEFAULT | CF_WRITE, &DataType::Index) => {
+                    // No need to write `write_cf` and index data to csv file
+                    return Ok(());
+                }
+                (..) => unreachable!(),
```
Comment (NingLin-P) [inline components/backup_text/src/rwer.rs]: ditto

File: components/backup_text/src/bin/rewrite/br_models.rs (hunk 1 above)
Comment (NingLin-P) [reply components/backup_text/src/bin/rewrite/br_models.rs:16]: I means `name: String` 🤣

**Review by NingLin-P — COMMENTED** [review]
Rest LGTM

File: components/backup_text/src/bin/rewrite/br_models.rs (hunk 1 above)
Comment (BugenZhao) [reply components/backup_text/src/bin/rewrite/br_models.rs:16]: We must keep the structure consistent with models from `br`. 🥺

**Review by NingLin-P — APPROVED** [review]

**Pull request merged**